library/spdm_crypt_lib/fips/libspdm_selftest_hmac.c

```diff
@@ -0,0 +1,108 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2023 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#include "internal/libspdm_crypt_lib.h"
+
+#if LIBSPDM_FIPS_MODE
+
+/**
+ * HMAC KAT test: HMAC-SHA512 KAT and HMAC-SHA384 KAT and HMAC-SHA256 KAT.
+ * HMAC-SHA512 KAT covers SHA384, SHA512, HMAC-SHA384, and HMAC-SHA512.
+ * HMAC-SHA384 KAT covers SHA384, SHA512, HMAC-SHA384, and HMAC-SHA512.
+ * HMAC-SHA256 KAT covers SHA1, SHA224, SHA256, HMAC-SHA1, HMAC-SHA224, and HMAC-SHA256.
+ **/
+bool libspdm_fips_selftest_hmac(void)
+{
+    bool result;
+    bool status;
+
+    status = true;
+    result = true;
+
+#if (LIBSPDM_SHA256_SUPPORT || LIBSPDM_SHA384_SUPPORT || LIBSPDM_SHA512_SUPPORT)
+    const uint8_t key[32] = {0};
+    const uint8_t msg[32] = {0};
+#endif/*(LIBSPDM_SHA256_SUPPORT || LIBSPDM_SHA384_SUPPORT || LIBSPDM_SHA512_SUPPORT)*/
+
+#if LIBSPDM_SHA256_SUPPORT
+    uint8_t hmac_256_result[32];
+    const uint8_t hmac_sha256_answer[32] = {
+        0x33, 0xad, 0x0a, 0x1c, 0x60, 0x7e, 0xc0, 0x3b,
+        0x09, 0xe6, 0xcd, 0x98, 0x93, 0x68, 0x0c, 0xe2,
+        0x10, 0xad, 0xf3, 0x00, 0xaa, 0x1f, 0x26, 0x60,
+        0xe1, 0xb2, 0x2e, 0x10, 0xf1, 0x70, 0xf9, 0x2a
+    };
+    libspdm_zero_mem(hmac_256_result, sizeof(hmac_256_result));
+    result = libspdm_hmac_sha256_all(msg, sizeof(msg), key, sizeof(key), hmac_256_result);
+    if (!result) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "gen hmac_sha256 failed \n"));
+        status = false;
+    }
+
+    if (libspdm_const_compare_mem(hmac_256_result, hmac_sha256_answer,
+                                  sizeof(hmac_sha256_answer))) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "hmac_sha256 KAT failed \n"));
+        status = false;
+    }
+
+#endif/*LIBSPDM_SHA256_SUPPORT*/
+
+#if LIBSPDM_SHA384_SUPPORT
```

Comment:
Make this a separate function.

Reply:
Thanks. I have made three separate function.